datasources/crafts.js

```diff
@@ -1,48 +1,86 @@
-// datasource for crafts
+// datasource for crafts 
+class CraftsAPI {
+    constructor(){
+        this.caache = false;
```

Comment:
```suggestion
        this.cache = false;
```

This looks like it might be a typo

Reply:
Thanks, fixed!